interfaces/prompting/patterns/patterns_test.go

```diff
@@ -864,3 +864,644 @@ func (s *patternsSuite) TestPathPatternMatchesErrors(c *C) {
+				"/foo/bar*/**/baz",
+			},
+			"/foo/bar/**/*baz",
+		},
+		{
+			"/foo/bar/fizz/buzz/baz/",
+			[]string{
+				"/foo/bar/**/",
+				"/foo/bar/**",
+			},
+			"/foo/bar/**/",
+		},
+		{
+			"/foo/bar/fizz/buzz/baz/",
+			[]string{
+				"/foo/bar/**/",
+				"/foo/bar*/**/baz/",
+			},
+			"/foo/bar/**/",
+		},
+		{
+			"/foo/bar/fizz/buzz/baz/",
+			[]string{
+				"/foo/bar/**",
+				"/foo/bar*/**/baz/",
+			},
+			"/foo/bar/**",
+		},
+		{
+			"/foo/bar/fizz/buzz/baz/",
+			[]string{
+				"/foo/bar*/**/baz",
+				"/foo/bar*/**/",
+			},
+			"/foo/bar*/**/baz",
+		},
+		{
+			"/foo/barfizz/buzz/baz/",
+			[]string{
+				"/foo/bar*/**/",
+				"/foo/bar*/**",
+			},
+			"/foo/bar*/**/",
+		},
+		{
+			"/foo/bar/file.tar.gz",
+			[]string{
+				"/foo/bar/*.gz",
+				"/foo/bar/*.tar.gz",
+			},
+			"/foo/bar/*.tar.gz",
+		},
+		{
+			"/foo/bar/file.tar.gz",
+			[]string{
+				"/foo/bar/**/*.gz",
+				"/foo/**/*.tar.gz",
+			},
+			"/foo/bar/**/*.gz",
+		},
+		{
+			"/foo/bar/x/y/z/file.tar.gz",
+			[]string{
+				"/foo/bar/x/**/*.gz",
+				"/foo/bar/**/*.tar.gz",
+			},
+			"/foo/bar/x/**/*.gz",
+		},
+		{
+			"/foo/bar/file.tar.gz",
+			[]string{
+				"/foo/bar/**/*.tar.gz",
+				"/foo/bar/*",
+			},
+			"/foo/bar/*",
+		},
+		{
+			"/foo/bar/baz/x/y/z/file.txt",
+			[]string{
+				"/foo/bar/**",
+				"/foo/bar/baz/**",
+				"/foo/bar/baz/**/*.txt",
+			},
+			"/foo/bar/baz/**/*.txt",
+		},
+		{
+			"/foo/bar",
+			[]string{
+				"/foo/bar*",
+				"/foo/bar/**",
+			},
+			"/foo/bar/**",
+		},
+		{
+			`/foo/\`,
+			[]string{
+				`/foo/\\`,
+				`/foo/*/**`,
+			},
+			`/foo/\\`,
+		},
+		{
+			`/foo/*fizz/bar/x*`,
+			[]string{
+				`/foo/\**/b\ar/*\*`,
+				`/foo/*/bar/x\*`,
+			},
+			`/foo/\**/bar/*\*`,
+		},
+		{
+			"/foo/barxxxbaz",
+			[]string{
+				"/foo/bar**",
+				"/foo/bar**baz",
+			},
+			"/foo/bar*baz",
+		},
+		{
+			"/foo/xxxbar",
+			[]string{
+				"/foo/**",
+				"/foo/**bar",
+			},
+			"/foo/*bar",
+		},
+		{
+			"/foo/x/y/z/bar/baz/",
+			[]string{
+				"/foo/**/bar/*/",
+				"/foo/**/b?r/baz/",
+			},
+			"/foo/**/bar/*/",
+		},
+		{
+			"/foo/x/y/z/bar/fizz",
+			[]string{
+				"/foo/**/fizz",
+				"/foo/**/bar/fizz",
+			},
+			"/foo/**/bar/fizz",
+		},
+		{
+			"/foo/x/y/z/bar",
+			[]string{
+				"/foo/**/*/bar",
+				"/foo/**/*",
+			},
+			"/foo/*/**/bar",
+		},
+		{
+			"/foo/bar",
+			[]string{
+				"/foo/**/*",
+				"/**",
+			},
+			"/foo/**",
+		},
+		// Duplicate patterns should never be passed into HighestPrecedencePattern,
+		// but if they are, handle them correctly.
+		{
+			"/foo/bar/",
+			[]string{
+				"/foo/bar/",
+				"/foo/bar/",
+			},
+			"/foo/bar/",
+		},
+		{
+			"/foo/bar/",
+			[]string{
+				"/foo/bar/",
+				"/foo/bar/",
+				"/foo/bar",
+			},
+			"/foo/bar/",
+		},
+		{
+			"/foo/bar/baz/",
+			[]string{
+				"/foo/bar/**",
+				"/foo/bar/**",
+				"/foo/bar/*",
+			},
+			"/foo/bar/*",
+		},
+	} {
+		variants := make([]*patterns.PatternVariant, len(testCase.patterns))
```

Comment:
can you add a sanity check that calls PathPatternMatches for each of the patterns and the path?

Reply:
Yes, good idea!